Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -12,311 +12,811 @@
+/// the extent of code context included with diagnostic messages, enhancing its utility in debugging processes.
+///
+/// ### Example
+///
+/// ```swift
+/// // Swift source code with issues
+/// class Iterator: {
+///   var counter = 0
+///   var wasCounterCalled: Bool { counter > 0 }
+///
+///   init(counter: Int = 0) {
+///     self.counter = counter
+///   }
+///
+///   // Description of the `loopOver` function
+///   func loopOver() {
+///     for (i = 0; i != 10; i += 1) { }
+///   }
+/// }
+///
+/// // Diagnostics produced by a parser could be applied to produce an annotated output like:
+/// """
+///  1 │ class Iterator: {
+///    │                 ╰─ error: expected type in inherited type
+///  2 │   var counter = 0
+///  3 │   var wasCounterCalled: Bool { counter > 0 }
+///    ┆
+///  9 │   // Description of the `loopOver` function
+/// 10 │   func loopOver() {
+/// 11 │     for (i = 0; i != 10; i += 1) { }
+///    │         ~~~~~~~~~~~~~~~~~~~~~~~~
+///    │     │   │  ╰─ error: expected ')' to end tuple pattern
+///    │     │   ╰─ note: to match this opening '('
+///    │     ╰─ error: C-style for statement has been removed in Swift 3
+/// 12 │   }
+/// 13 │ }
+/// """
+/// ```
+///
+/// ### Motivation
+///
+/// This formatter enables better development practices by:
+///
+/// - **Improving Debugging**: Diagnostic messages, including errors, warnings, notes, fix-its, highlights, and remarks,
+/// are aligned in context to the source code, making it easier to understand what needs to be corrected or considered.
+///
+/// - **Enhanced Readability**: The formatter enriches the source code with line numbers,
+/// annotations, and optional text suffixes, providing a full picture to the developer.
+///
+/// - **Flexible Customization**: Through the `DiagnosticDecorator` protocol, you can customize
+/// the appearance of diagnostic output.
 public struct DiagnosticsFormatter {
-
-  /// A wrapper struct for a source line, its diagnostics, and any
-  /// non-diagnostic text that follows the line.
-  private struct AnnotatedSourceLine {
-    var diagnostics: [Diagnostic]
-    var sourceString: String
-
-    /// Non-diagnostic text that is appended after this source line.
-    ///
-    /// Suffix text can be used to provide more information following a source
-    /// line, such as to provide an inset source buffer for a macro expansion
-    /// that occurs on that line.
-    var suffixText: String
-
-    /// Whether this line is free of annotations.
-    var isFreeOfAnnotations: Bool {
-      return diagnostics.isEmpty && suffixText.isEmpty
-    }
+  /// An enumeration that describes the amount of contextual lines to display around each diagnostic message.
+  ///
+  /// - `limited(Int)`: Display a limited number of lines around each diagnostic. The integer value specifies the number of lines before and after the diagnostic message.
```

Comment:
I know that swift-format doesn’t complain about overly long comment lines but would be good to manually break these lines so they are < 160 columns

---

Also, I think the doc comments for the enum cases should be put on those cases instead of on the `ContextRange` enum itself.

Reply:
I've moved documentation to each enum case + I've ensured that every line of documentation is shorter than 120 columns.